File: app/src/main/java/com/projectreachout/Article/GetArticle/Article.java
package com.projectreachout.Article.GetArticle;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.projectreachout.User.User;

public class Article {
    private String article_id;
    private String user_id;
    private String display_name;
    private String time_stamp;
    private String avatar;
    private String image_url;
    private String description;
    private String my_reaction;
    private String reaction_count;

    public Article() {
    }

    public static Article fromJson(String jsonString) {
        return new Gson().fromJson(jsonString, Article.class);
    }

    public String getId() {
        return article_id;
    }

    public String getDisplay_name() {
        return display_name;
    }

    public String getTime_stamp() {
        return time_stamp;
    }

    public String getAvatar() {
        return avatar;
    }

    public String getImage_url() {
        return image_url;
    }

    public String getDescription() {
        return description;
    }

    public void setId(String article_id) {
        this.article_id = article_id;
    }

    public void setDisplay_name(String display_name) {
        this.display_name = display_name;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getArticle_id() {
        return article_id;
    }

    public String getUser_id() {
        return user_id;
    }

    public void setUser_id(String user_id) {
        this.user_id = user_id;
    }

    public String getMy_reaction() {
        return my_reaction;
    }

    public String getReaction_count() {
        return reaction_count;
    }

    @NonNull
    @Override
    public String toString() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }

    public static class Reaction {
        private User user;
        private String reaction;

        public static Reaction fromJSON(String jsonString) {
            return new Gson().fromJson(jsonString, Reaction.class);
        }

        public User getUser() {
            return user;
        }

        public String getReaction() {
            return reaction;
        }
    }
}

File: app/src/main/java/com/projectreachout/v3/models/Article.kt
package com.projectreachout.v3.models

import com.google.firebase.firestore.DocumentSnapshot
import com.google.gson.Gson
import com.google.gson.annotations.SerializedName
import java.util.*

class Article {
    companion object {
        fun fromDoc(doc: DocumentSnapshot): Article {
            val userEntity: ArticleEntity? = doc.toObject(ArticleEntity::class.java)
            return Gson().fromJson(Gson().toJson(userEntity), Article::class.java)
        }
    }

    var articleId: String? = null
    var userId: String? = null
    var timeStamp: Date? = null
    var imageUrl: String? = null
    var description: String? = null
    var reactionCount: Int? = null

    private class ArticleEntity {
        @SerializedName("articleId")
        var article_id: String? = null

        @SerializedName("userId")
        var user_id: String? = null

        @SerializedName("timeStamp")
        var time_stamp: Date? = null

        @SerializedName("imageUrl")
        var image_url: String? = null
        var description: String? = null

        @SerializedName("reactionCount")
        var reaction_count: Int? = null
    }
}